src/test/java/org/jenkinsci/plugins/ewm/steps/ExwsAllocateStepTest.java

```diff
@@ -0,0 +1,205 @@
+        j.assertLogContains(format("Physical path on disk was not provided in the Jenkins global config for Disk ID: '%s', within Disk Pool ID '%s'", DISK_ID, DISK_POOL_ID), upstreamRun);
+    }
+
+    @Test
+    public void physicalPathOnDiskNotRelative() throws Exception {
+        setupDiskPool(new Disk(DISK_ID, "name", "mount", "/path"));
+        createUpstreamJobAndRun();
+
+        j.assertBuildStatus(FAILURE, upstreamRun);
+        j.assertLogContains(format("Physical path on disk defined for Disk ID '%s', within Disk Pool ID '%s' must be a relative path", DISK_ID, DISK_POOL_ID), upstreamRun);
+    }
+
+    @Test
+    public void successfullyAllocateWorkspace() throws Exception {
+        Disk disk = new Disk(DISK_ID, "name", "mount", "path");
+        setupDiskPool(disk);
+        createUpstreamJobAndRun();
+
+        j.assertBuildStatusSuccess(upstreamRun);
+        j.assertLogContains(format("Selected Disk ID '%s' from the Disk Pool ID '%s'", DISK_ID, DISK_POOL_ID), upstreamRun);
+        j.assertLogContains(format("The path on Disk is: %s/%s/%d", disk.getPhysicalPathOnDisk(), upstreamRun.getParent().getFullDisplayName(), upstreamRun.getNumber()), upstreamRun);
+    }
+
+    /* ##### Tests for the downstream Job ###### */
+
+    @Test
+    public void downstreamJobWithInvalidUpstreamName() throws Exception {
+        String upstreamName = "random-upstream-name";
+        createDownstreamJobAndRun(upstreamName);
+
+        j.assertBuildStatus(FAILURE, downstreamRun);
+        j.assertLogContains(format("There isn't any upstream job associated with '%s'", upstreamName), downstreamRun);
+    }
+
+    @Test
+    public void upstreamJobHasNoStableBuild() throws Exception {
+        createUpstreamJobAndRun("random-disk-pool");
+        String upstreamName = upstreamRun.getParent().getName();
+        createDownstreamJobAndRun(upstreamName);
+
+        j.assertBuildStatus(FAILURE, upstreamRun);
+        j.assertBuildStatus(FAILURE, downstreamRun);
+        j.assertLogContains(format("'%s' doesn't have any stable build", upstreamName), downstreamRun);
+    }
+
+    @Test
+    public void upstreamJobHasNoActionRegistered() throws Exception {
+        String jobName = "test";
+        createWorkflowJobAndRun(jobName, "echo 'hello world'");
+        createDownstreamJobAndRun(jobName);
+
+        j.assertBuildStatus(FAILURE, downstreamRun);
+        j.assertLogContains(format("The Jenkins job '%s' does not have registered any 'External Workspace Allocate' action", jobName), downstreamRun);
+    }
+
+    @Test
+    public void upstreamBuildIsNotPipelineJob() throws Exception {
+        String jobName = "name";
+        FreeStyleBuild build = j.createFreeStyleProject(jobName).scheduleBuild2(0).get();
+        createDownstreamJobAndRun(jobName);
+
+        j.assertBuildStatus(FAILURE, downstreamRun);
+        j.assertLogContains(format("Build '%s' is not a Pipeline job. Can't read the run actions", build), downstreamRun);
```

Comment:
Not important: I can't seem to find the code that causes this error message in the plugin code, so I guess it comes from the pipeline code?